driver-core/src/main/com/mongodb/connection/DefaultClusterFactory.java

```diff
@@ -94,28 +103,72 @@ public Cluster create(final ClusterSettings settings, final ServerSettings serve
+     * @param heartbeatStreamFactory the heartbeat stream factory
+     * @param credentialList         the credential list
+     * @param commandListener        an optional listener for command-related events
+     * @param applicationName        an optional application name to associate with connections to the servers in this cluster
+     * @param mongoDriverInformation the optional driver information associate with connections to the servers in this cluster
+     * @return the cluster
+     *
+     * @since 3.5
+     */
+    public Cluster create(final ClusterSettings clusterSettings, final ServerSettings serverSettings,
+                          final ConnectionPoolSettings connectionPoolSettings, final StreamFactory streamFactory,
+                          final StreamFactory heartbeatStreamFactory,
+                          final List<MongoCredential> credentialList,
+                          final CommandListener commandListener,
+                          final String applicationName,
+                          final MongoDriverInformation mongoDriverInformation) {
+        return create(clusterSettings, serverSettings, connectionPoolSettings, streamFactory, heartbeatStreamFactory, credentialList,
+                getClusterListener(clusterSettings.getClusterListeners()),
+                getConnectionPoolListener(connectionPoolSettings.getConnectionPoolListeners()),
+                getServerMonitorListener(serverSettings.getServerMonitorListeners()),
+                commandListener, applicationName, mongoDriverInformation);
+    }
+
+    Cluster create(final ClusterSettings clusterSettings, final ServerSettings serverSettings,
+                          final ConnectionPoolSettings connectionPoolSettings, final StreamFactory streamFactory,
+                          final StreamFactory heartbeatStreamFactory,
+                          final List<MongoCredential> credentialList,
+                          final ClusterListener clusterListener,
+                          final ConnectionPoolListener connectionPoolListener,
+                          final ServerMonitorListener serverMonitorListener,
+                          final CommandListener commandListener,
+                          final String applicationName,
+                          final MongoDriverInformation mongoDriverInformation) {
+
+        ClusterId clusterId = new ClusterId(clusterSettings.getDescription());
+        ClusterableServerFactory serverFactory = new DefaultClusterableServerFactory(clusterId, clusterSettings, serverSettings,
                 connectionPoolSettings, streamFactory, heartbeatStreamFactory, credentialList,
-                connectionListener != null ? connectionListener : new NoOpConnectionListener(),
-                connectionPoolListener != null ? connectionPoolListener : new NoOpConnectionPoolListener(),
+                getConnectionPoolListenerOrDefault(connectionPoolListener),
```

Comment:
Same issue as I called out in BaseCluster. Rather than pass ConnectionPoolListener twice, let's merge the list from ConnectionPoolSettings with any that is passed in and create a new ConnectionPoolSettings with the merged list.